apps/language_server/lib/language_server/experimental/provider/code_action/replace_remote_function.ex

```diff
@@ -0,0 +1,147 @@
+defmodule ElixirLS.LanguageServer.Experimental.Provider.CodeAction.ReplaceRemoteFunction do
+  @moduledoc """
+  Code actions that replace unknown remote function with ones suggested by the warning message
+  """
+
+  alias ElixirLS.LanguageServer.Experimental.CodeMod
+  alias ElixirLS.LanguageServer.Experimental.CodeMod.Ast
+  alias ElixirLS.LanguageServer.Experimental.Protocol.Types.CodeAction, as: CodeActionResult
+  alias ElixirLS.LanguageServer.Experimental.Protocol.Types.Diagnostic
+  alias ElixirLS.LanguageServer.Experimental.Protocol.Types.TextEdit
+  alias ElixirLS.LanguageServer.Experimental.Protocol.Types.Workspace
+  alias ElixirLS.LanguageServer.Experimental.SourceFile
+
+  @pattern ~r/(.*)\/(.*) is undefined or private. .*:\n(.*)/s
+
+  @spec pattern() :: Regex.t()
+  def pattern, do: @pattern
+
+  @spec apply(SourceFile.t(), Diagnostic.t()) :: [CodeActionResult.t()]
+  def apply(source_file, diagnostic) do
+    with {:ok, module, name} <- extract_function(diagnostic.message),
+         {:ok, suggestions} <- extract_suggestions(diagnostic.message),
+         one_based_line = extract_line(diagnostic),
```

Comment:
nit: I'm not a big fan of putting things in a with clause that can't fail, you don't have any guarantees about what `one_based_line` can be here, it can be nil, for example.

Reply:
Actually, I'm matching against the `Types.Diagnostic` module in the `extract_line` function and it seems like none of the keys in `diagnostic.range.start.line` have an optional type, so why isn't it guaranteed that `one_based_line` is an integer?

Also, I think changing `extract_line` to

```elixir
defp extract_line(%Diagnostic{} = diagnostic) do
  {:ok, diagnostic.range.start.line}
end
```
does not give us anything. If it is not guaranteed that `diagnostic.range.start.line` can't be nil, we would just get `{:ok, nil}` in `apply` function and still have the same problem.

By `one_based_line = extract_line(diagnostic),` I'm avoiding nested `with`s.